app/controllers/incomeSources/add/CheckUKPropertyDetailsController.scala

```diff
@@ -97,23 +100,42 @@ class CheckUKPropertyDetailsController @Inject()(val checkUKPropertyDetails: Che
         }
   }
 
-  def handleRequest(isAgent: Boolean)(implicit user: MtdItUser[_]): Future[Result] = {
+  def handleRequest(isAgent: Boolean)(implicit user: MtdItUser[_]): Future[Result] = withIncomeSourcesFS {
     val backUrl = getBackUrl(isAgent)
     val postAction = getSubmitUrl(isAgent)
     val errorHandler = getErrorHandler(isAgent)
 
-    withIncomeSourcesFS {
-      getUKPropertyDetailsFromSession(sessionService)(user, ec).map(_.toOption).map {
-        case Some(checkUKPropertyViewModel: CheckUKPropertyViewModel) =>
-          Ok(
-            checkUKPropertyDetails(viewModel = checkUKPropertyViewModel,
-              isAgent = isAgent,
-              backUrl = backUrl,
-              postAction = postAction))
-        case None => Logger("application").error(
-          s"[CheckUKPropertyDetailsController][handleRequest] - Error: Unable to build UK property details")
-          errorHandler.showInternalServerError()
-      }
+    getUKPropertyDetailsFromSession(user, ec).map(_.toOption).map {
+      case Some(checkUKPropertyViewModel: CheckUKPropertyViewModel) =>
+        Ok(
+          checkUKPropertyDetails(viewModel = checkUKPropertyViewModel,
+            isAgent = isAgent,
+            backUrl = backUrl,
+            postAction = postAction))
+      case None => Logger("application").error(
+        s"[CheckUKPropertyDetailsController][handleRequest] - Error: Unable to build UK property details")
+        errorHandler.showInternalServerError()
+    }
+  }
+
+  def getUKPropertyDetailsFromSession(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Either[Throwable, CheckUKPropertyViewModel]] = {
+    for {
+      startDate <- sessionService.getMongoKeyTyped[LocalDate](dateStartedField, JourneyType(Add, UkProperty))
+      accMethod <- sessionService.getMongoKeyTyped[String](incomeSourcesAccountingMethodField, JourneyType(Add, UkProperty))
+    } yield (startDate, accMethod) match {
+      case (Right(dateMaybe), Right(methodMaybe)) =>
+        val maybeModel = for {
+          ukPropertyStartDate <- dateMaybe
+          cashOrAccrualsFlag <- methodMaybe
+        } yield {
+          CheckUKPropertyViewModel(
+            tradingStartDate = ukPropertyStartDate,
+            cashOrAccrualsFlag = cashOrAccrualsFlag)
+        }
+        maybeModel.map(Right(_))
+          .getOrElse(Left(new Error("Unable to construct UK property view model")))
+      case (_, _) =>
+        Left(new Error("Error occurred when retrieving start dat and accounting method from session storage"))
```

Comment:
Could fix: errors relate to JVM (typically cannot be recovered from), these should be exceptions as we are handling them
Also typo on line 138